src/App.tsx

```diff
@@ -1,61 +1,116 @@
 import './App.scss';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUser(UserId: number) {
+  const foundUser = usersFromServer.find(user => user.id === UserId);
+
+  return foundUser || null;
+}
+
+const todos: Todo[] = todosFromServer.map((todo) => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [userId, setUserId] = useState(0);
+  const [userIdError, setUserIdError] = useState(false);
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState(false);
+
+  const changeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setTitleError(false);
+    setTitle(value);
+  };
+
+  const changeUserId = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    setUserIdError(false);
+    setUserId(Number(value));
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={(event) => {
+          event.preventDefault();
+
+          const lastId = Math.max(0, ...todos.map(({ id }) => id));
+
+          const newTodo: Todo = {
+            id: lastId + 1,
+            userId,
+            title,
+            completed: false,
+            user: getUser(userId),
+          };
+
+          if (userId && title.trim()) {
+            todos.push(newTodo);
+          } else {
+            setTitleError(title.trim() === '');
+            setUserIdError(userId === 0);
+
+            if (title.trim() === '') {
+              setTitle('');
+            }
+
+            return;
+          }
+
+          setTitle('');
+          setUserId(0);
+        }}
```

Comment:
Create separate methods instead of creating inline event handlers.
<img width="1058" alt="Screenshot 2022-09-14 at 12 19 46" src="https://user-images.githubusercontent.com/17098736/190176837-ad56985e-a3dd-4046-bc3f-372f7bade9a5.png">